torchserve: add container-based config discovery support (#24609)
DataDog/integrations-core: torchserve/README.md

Follow the instructions below to install and configure this check for an Agent running on a host. For containerized environments, see the [Autodiscovery Integration Templates][3] for guidance on applying these instructions.

> #### Container-based configuration discovery

Reviewer: ```suggestion
### Container-based configuration discovery
```

> This check supports [Autodiscovery][3] based on containers exposing TorchServe's OpenMetrics endpoint (default port `8082`). Discovery only generates an `openmetrics_endpoint` instance; the Inference API and Management API endpoints cannot be derived automatically and remain manually configurable, as described below. Discovery also requires `metrics_mode` to be set to `prometheus` in `config.properties`, since TorchServe defaults to `log` mode, which the Agent cannot scrape.

Reviewer: ```suggestion
This check supports [Autodiscovery][3] based on containers exposing TorchServe's OpenMetrics endpoint (default port `8082`). Discovery only generates an `openmetrics_endpoint` instance; Autodiscovery cannot derive the Inference API and Management API endpoints, configure those manually using the steps in the [Configuration](#configuration) section. Discovery also requires `metrics_mode` to be set to `prometheus` in `config.properties`, since TorchServe defaults to `log` mode, which the Agent cannot scrape.
```
